Add io.itch.tx00100xt.SeriousSamClassic-VK

tx00100xt opened the pull request:
Good day everyone.

Before creating a pull request https://github.com/flathub/flathub/pull/4885
I used org.zdoom.GZDoom and org.dhewm3.Dhewm3 as templates to create a flatpack. 
It is always convenient to have a ready-made example. Similarly, I used the package name org.serioussam.SeriousSamClassic-VK. 
I didn't think there should be a serioussam.org site because I had never used a flatpack before or created a flatpack package. 
This is also why I added README.md, org.serioussam.SeriousSamClassic-VK.64.png, org.serioussam.SeriousSamClassic-VK.48.png because they were in the specified packages.

Now the package name is io.itch.tx00100xt.SeriousSamClassic-VK. The address https://tx00100xt.itch.io/ exists. 
I also took into account the comments regarding corrections in the pull request: https://github.com/flathub/flathub/pull/4885

I decided to continue creating the flatpack package because I was asked: https://github.com/tx00100xt/SeriousSamClassic-VK/issues/36
And because it’s easier for me, as the owner of the repository, to do this.
Let's be mutually polite. If anything needs to be changed in this pull request, please write.

I added patches for flatpack: https://github.com/tx00100xt/SeriousSamClassic-VK/commit/775f17d6e88d44c6b052206760b6f49b60cd83a2
and made a new tag for it.
The build and run has been tested on x86_64, but in theory it should work on different architectures:
https://github.com/tx00100xt/SeriousSamClassic-VK/actions/runs/7653800491

### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

hfiguiere commented on io.itch.tx00100xt.SeriousSamClassic-VK.yaml:
```diff
@@ -0,0 +1,57 @@
+        dest-filename: 0003-desktop-file.patch
+      - type: shell
+        commands:
+          - patch -p1 < 0001-CMakeLists.txt.patch
+          - patch -p1 < 0002-Engine.cpp.patch
+          - patch -p1 < 0003-desktop-file.patch
```
this is not how patches are handled in flatpak.

tx00100xt replied:
I made local patches first. As stated here:
https://docs.flatpak.org/en/latest/manifests.html#
>- type: patch
>  path:  0001-CMakeLists.txt.patch
>- type: patch
>  path:  0002-Engine.cpp.patch
>- type: patch
>  path:  0003-desktop-file.patch
>

Here I didn’t find how to use remote patches
https://docs.flatpak.org/en/latest/flatpak-builder-command-reference.html

I think we can add patches to pull requests.
Here's how it's done:
https://github.com/flathub/org.gnome.Dictionary/blob/master/0001-fix-image-urls.patch

bbhtt replied:
Add them as `type: patch`

tx00100xt commented:
I updated the pull request. If anything else needs to be corrected please write.

hfiguiere commented on io.itch.tx00100xt.SeriousSamClassic-VK.yaml:
```diff
@@ -0,0 +1,46 @@
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=RelWithDebInfo
+    sources:
+      - type: archive
+        url: https://github.com/tx00100xt/SeriousSamClassic-VK/archive/refs/tags/1.10.6e.tar.gz
+        sha256: aec1d1ef3a16bb914d378d8b777d8178341253994d0e8080a514a67814039bce
+        x-checker-data:
+          type: anitya
+          project-id: 9154
+          url-template: https://github.com/tx00100xt/SeriousSamClassic-VK/archive/refs/tags/$version.tar.gz
+      - type: patch
+        path: 0001-CMakeLists.txt.patch
+      - type: patch
+        path: 0002-Engine.cpp.patch
+      - type: patch
+        path: 0003-desktop-file.patch
+
+  - name: metainfo
+    buildsystem: simple
+    sources:
+      - type: file
+        path: io.itch.tx00100xt.SeriousSamClassic-VK.metainfo.xml
+    build-commands:
+      - install -Dm 644 io.itch.tx00100xt.SeriousSamClassic-VK.metainfo.xml -t /app/share/metainfo
```
```suggestion
    post-install:
      - install -Dm 644 io.itch.tx00100xt.SeriousSamClassic-VK.metainfo.xml -t /app/share/metainfo
    sources:
      - type: archive
        url: https://github.com/tx00100xt/SeriousSamClassic-VK/archive/refs/tags/1.10.6e.tar.gz
        sha256: aec1d1ef3a16bb914d378d8b777d8178341253994d0e8080a514a67814039bce
        x-checker-data:
          type: anitya
          project-id: 9154
          url-template: https://github.com/tx00100xt/SeriousSamClassic-VK/archive/refs/tags/$version.tar.gz
      - type: patch
        path: 0001-CMakeLists.txt.patch
      - type: patch
        path: 0002-Engine.cpp.patch
      - type: patch
        path: 0003-desktop-file.patch
      - type: file
        path: io.itch.tx00100xt.SeriousSamClassic-VK.metainfo.xml
```

hfiguiere replied:
also I doubt the project-id is correct.

![Screenshot from 2024-01-25 12-13-15](https://github.com/flathub/flathub/assets/114441/e1c0184b-dee7-4bee-a964-1253432158f9)

tx00100xt commented:
Thanks for your comments and corrections. Yes, this is left over from the Dhewm3 manifest. I updated the pull request. Please write if anything else needs to be corrected.

hfiguiere commented:
bot, build io.itch.tx00100xt.SeriousSamClassic-VK

hfiguiere commented:
There is no desktop file installed with a name that matches the app id + `.desktop`.

This was mentionned there:
https://github.com/flathub/flathub/pull/4885#discussion_r1463267093

hfiguiere commented on io.itch.tx00100xt.SeriousSamClassic-VK.metainfo.xml:
```diff
@@ -0,0 +1,73 @@
+    <release version="1.10.6e" date="2024-01-25"/>
+    <release version="1.10.6d" date="2024-01-17"/>
+  </releases>
+
```
also this fail to validate:

```
        "tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]"
```

tx00100xt replied:
Latest release tagged 1.10.6d
1.10.6e tag only.  I'll fix it now

tx00100xt commented:
I updated the pull request. Now the names of the desktop files:
>io.itch.tx00100xt.SeriousSamClassic-VK.desktop
>io.itch.tx00100xt.SeriousSamClassic-VK.SE.desktop
>
Corrected for the last two releases
`<release version="1.10.6d" date="2024-01-17"/>`
`<release version="v1.10.6" date="2023-10-16"/>`
>
https://github.com/tx00100xt/SeriousSamClassic-VK/releases

bbhtt commented:
bot, build io.itch.tx00100xt.SeriousSamClassic-VK

tx00100xt commented:
>    "errors": [
        "desktop-file-icon-key-wrong-value",
        "appstream-missing-icon-file",
        "appstream-failed-validation"
>
I guess I need to give the name of the file icons "application name + extension(.png)"
>"tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]"
>
The metadata file is missing the content_rating section

tx00100xt commented:
I updated the pull request. Changed the name of the file icons and added a content_rating section.
Check please.

tx00100xt commented:
I added an icon size 128x128. I found an error in the metadata file (the desktop file name was incorrect). Changed the description of the screenshots. Please check.

bbhtt commented:
bot, build io.itch.tx00100xt.SeriousSamClassic-VK

bbhtt commented:
The first location in both dialogues is a bit misleading, unless you ship it with the flatpak it cannot be put in `/app`, it is read-only. So the only path users can put game data is in `~/.var/app/app-id/...`

![image](https://github.com/flathub/flathub/assets/62639087/0f389339-174e-4431-8fe0-bdc9a5f6dc8e)

tx00100xt commented:
I'll fix it now. I just need to change the patch 0002-Engine.cpp.patch a little.

tx00100xt commented:
I corrected the message to leave only one path in the home directory. Check please.

tx00100xt commented:
Doesn't work on KDE. Complete in a few seconds. It works fine on the mate. Yesterday it worked fine on KDE too. I need to figure out what the problem is. Moreover, the old version with the name org,serioussam works fine on the KDE. I will update the pull request when I solve the problem

tx00100xt commented:
Maybe I'm the only one with a problem. Now even building the game using the flatpack-builder is interrupted every 5-6 seconds in KDE. Everything is fine in Mate.
I changed the application name to io.itch.tx00100xt.SeriousSam, and now the game builds again and works fine in KDE Wayland and Mate X11. With the name io.itch.tx00100xt.SeriousSamClassic-VK in KDE Wayland/X11 does not compile (compilation is interrupted every 5-6 seconds) and the game also starts for 5-6 seconds and is interrupted.

Can I change the package name to io.itch.tx00100xt.SeriousSam? With this name everything works well in KDE and Mate.
Please tell me. If it is possible to change the name to a new one, I will update the pull request.

bbhtt commented:
The id should not have any effect on flatpak-builder.

tx00100xt commented:
> The id should not have any effect on flatpak-builder/building process.
> 
> What do you mean by interrupted?
> 
When I run the build command in KDE in the console:
```bash
flatpak-builder --force-clean --sandbox --user --install --install-deps-from=flathub --ccache --mirror-screenshots-url=https://dl.flathub.org/repo/screenshots --repo=repo builddir io.itch.tx00100xt.SeriousSamClassic-VK.yaml
```
after a few seconds I get the error:
Error: module serioussam: Child process exited with code 137
If I run the build in the console in Mate, this error does not exist.

If I start the build in KDE with the command
```bash
flatpak-builder --force-clean --sandbox --user --install --install-deps-from=flathub --ccache --mirror-screenshots-url=https://dl. flathub.org/repo/screenshots --repo=repo builddir io.itch.tx00100xt.SeriousSam.yaml
```
Everything is going well. And the game works correctly.

I myself cannot understand why the name affects the build and the launch of the game in the KDE.
> You can change it if you want but seems unnecessary.
>
Thanks I got it.
> You don't need to make another PR if you change, rename everything here.
>
I wanted to make changes to this pull request here, and I didn't want to create a new PR.

tx00100xt commented:
I won't change anything for now. Maybe I'm the only one with a problem.
Maybe someone can check the operation of the application in KDE.
 In the last commit I updated the message box. Now the only message left is about the need for data in the home directory `~/.var/app/app-id/...`
Check please.

bbhtt commented:
That sometimes happens when `flatpak-builder` gets killed by the background portal. It should've been fixed already, but sometimes you need to explicitly grant it background permissions.

https://github.com/flatpak/xdg-desktop-portal/issues/478

tx00100xt commented:
Thanks for the explanation. I noticed that a small window "background activity with the package name" appears in the at startup (KDE). And with the name "io.itch.tx00100xt.SeriousSamClassic-VK" this window has stopped appearing in the KDE.

tx00100xt commented:
I tested the package from the link:
```
flatpak install --user https://dl.flathub.org/build-repo/78647/io.itch.tx00100xt.SeriousSamClassic-VK.flatpakref
```

**Click on the image to watch the video (YouTube)**

[![Serious Sam Classic run on Flatpak](https://img.youtube.com/vi/3I8X7Z2gVQM/0.jpg)](https://www.youtube.com/watch?v=3I8X7Z2gVQM)

There is one point. After installation, the latest release version 1.10.6d is displayed, and the manifest uses the 1.10.6e tag.
Before creating the flatpack, I thought that it was possible to use remote patches. For this I made the tag 1.10.6e. Since the patches are in the pull request, I’m thinking of returning the tag 1.10.6d to the manifest.

tx00100xt commented:
I updated the pull request. Replaced 1.10.6e with 1.10.6d (Latest release). I rechecked the build and launch of the game from flatpak.

tx00100xt commented:
bot, build io.itch.tx00100xt.SeriousSamClassic-VK

barthalion commented:
/merge
